### ISSUE TITLE
Ability for NVDA to indicate spell error in real time.

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2024]
**Reported by aleskis on 2012-01-03 09:01**
An innovative functionality in comparaison with other screenreader should be to implement a specific sound to indicate spelling errors, in real time during the writting. By this way, the time for rereading could be largely reduced.

**Comment by nvaccessAuto** [comment]
Incubated in f73b35487457febdcfd2254e16f85172b8883401.

**Comment by nvaccessAuto** [comment]
Incubated in 7778f21910dc19b59a750cd6f74b80517d038c01.

**Comment by derekriemer** [comment]
Hi,
Could _reportErrorInPreviousWord be set to return a Boolean? I am tempted to write an addon that would use this function (monkey patch it) and if a spelling error was found, I would keep a note of it, so that  I could give users a report of their spelling error statistics, and maybe even write a game to improve spelling in real time. For example, you could open a dialog, and it would tell you your percentage of words incorrect, the total number of spelling errors ever, the number in this session of NVDA, whether you have spelled more words correctly than normal, and play a tone when you have spelled more words incorrectly than average.
I totally understand if this is a resounding No!

**Comment by jcsteh** [comment]
The function actually does the reporting, not just the checking, so you 
couldn't just call it from your own event. How/from where were you 
planning on calling it?

**Comment by derekriemer** [comment]
If possible I was going to monkeypatch it, and see if it returned true, 
I haven't studied this in detail yet. Is that not possible?
It would be like
def _reportErrorInPreviousWord(self):
     if _reportErrorInPreviousWord _old(self):
         bla
     else
         bla

then I would assign that function in place of the one in the NVDAObject.

On 4/5/2016 4:03 PM, James Teh wrote:
> The function actually does the reporting, not just the checking, so you
> couldn't just call it from your own event. How/from where were you
> planning on calling it?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/2024#issuecomment-206002431>
>

-- 
------------------------------------------------------------------------


    Derek Riemer

  * Department of computer science, third year undergraduate student.
  * Proud user of the NVDA screen reader.
  * Open source enthusiast.
  * Member of Bridge Cu
  * Avid skiier.

Websites:
Honors portfolio <http://derekriemer.com>
Awesome little hand built weather app! 
<http://django.derekriemer.com/weather/>

email me at derek.riemer@colorado.edu <mailto:derek.riemer@colorado.edu>
Phone: (303) 906-2194

**Comment by jcsteh** [comment]
Ug. I think I'd prefer not to provide for a hack like that. The correct 
solution would be to allow add-ons to register for notifications about 
this, but I don't want to deal with that in this issue. :)

**Comment by derekriemer** [comment]
I totally understand. It's kind of an ugly hack. I just thought I would check before I look at other possible solutions like hacking envy wave or whatnot.

Sent from my heavily encrypted iPhone.
Please disregard errors as this is a smaller device.

**Comment by nvaccessAuto** [comment]
Incubated in af5c2bcce403331a0f8c987211f557a9a72cb97c.

**Comment by nvaccessAuto** [comment]
Incubated in 5720fc3b22291ba2db3b4de8d1428f5a660ea81a.
